src/routes/cities/components/CityMap.svelte

```diff
@@ -1,27 +1,104 @@
 <script lang="ts">
+	import mapboxgl, { type LngLatBoundsLike } from 'mapbox-gl';
+	import type { Popo } from 'src/types';
 	import { onMount } from 'svelte';
-	import mapboxgl from 'mapbox-gl';
+	import turf from 'turf';
+	import { location } from '../../../state';
 	mapboxgl.accessToken = import.meta.env.VITE_MAPBOX_ACCESS_TOKEN;
 
+	interface PopoLink {
+		url: string | null;
+		name: string;
+	}
 	let map: mapboxgl.Map;
+	export let popos: Popo[];
+	let markers: Array<mapboxgl.Marker> = [];
+	let popups: Array<mapboxgl.Popup> = [];
 
+	const features = turf.featureCollection(
+		popos.map((popo: Popo) =>
+			turf.point([popo.mapCoordinates[1], popo.mapCoordinates[0]], { name: popo.name })
+		)
+	);
+	const boundsArr = turf.bbox(features);
+	const mapBounds: LngLatBoundsLike = [boundsArr[0], boundsArr[1], boundsArr[2], boundsArr[3]];
+	const center = turf.center(features).geometry.coordinates;
 	onMount(() => {
 		map = new mapboxgl.Map({
 			container: 'map',
-			style: 'mapbox://styles/mapbox/light-v10',
-			center: [16.37, 48.2],
-			zoom: 12
+			style: 'mapbox://styles/mapbox/streets-v11',
+			center: [center[0], center[1]],
+			projection: { name: 'globe' }
+		});
+		addMarkers();
+		console.log(mapBounds);
+
+		setTimeout(() => map.fitBounds(mapBounds), 1000);
+	});
+
+	// need a better way to build popups
+	function addMarkers() {
+		popos.map((popo: Popo) => {
+			let imageUrl = `https://picsum.photos/seed/${popo.name}/600/450`;
+			if (popo.imageName) imageUrl = `/images/${popo.imageName}`;
+
+			const popup = new mapboxgl.Popup({
+				offset: 25,
+				closeButton: false,
+				className: 'bg-white rounded-md shadow-sm w-64'
+			}).setHTML(`
+      <div>
+      <img class="h-48 object-cover rounded-t-md w-full group-hover:scale-95 duration-150 group-hover:rounded-md"
+          alt=""
+          src="${imageUrl}">
+      </img>
+      <div class="p-2">
+      <div class="flex justify-center font-bold mt-2 text-base">${popo.name}</div>
+      <div class="mt-2 text-base">${popo.description}</div>
+      <div class="mt-2 text-base">${$location}</div>
+      </div>
+      </div>`);
+			popups.push(popup);
+			markers.push(
+				new mapboxgl.Marker()
+					.setLngLat([popo.mapCoordinates[1], popo.mapCoordinates[0]])
+					.setPopup(popup)
+					.addTo(map)
+			);
 		});
+	}
+
+	location.subscribe(() => {
```

Comment:
can create a [derived store](https://svelte.dev/tutorial/derived-stores) instead of subscribing